Steel update

Lycowolf opened the pull request:
#### Summary
Content "Steel update: vehicle parts"

#### Purpose of change
Use the new steel types for vehicle parts.

This is motivated by #80616 - I want to update all steel items in the game, this is a part of that effort.

#### Describe the solution
I changed the `steel` material to one of the new ones (`lc_steel`, `ch_steel` etc.). I also fixed several obvious errors (like electric motor not being made of copper or turbine engine being made of cast iron).

I made one commit per file. The rationale for choosing the specific materials is in the commit messages.

#### Describe alternatives you've considered
Updating other material than steel might be out of scope?


#### Testing
No testing yet.

Is there a way to test a lot of items at once?

Commit:
Steel update: wheels

Car axles: medium steel (https://www.iforgeiron.com/topic/4526-axles-what-metal/)
Car wheel rims: couldn't find, but presume same as axles (similar load
type)
Rail wheels: medium steel (https://www.kirail.com/Rail-Wheels/68.html)
Skateboard wheels: these can be made out of aluminum, i. e. mild steel
would be enough
Caster wheels etc.: cheap, low loading: mild steel

Commit:
Steel update: utilities

Most parts are not load bearing, i. e. mild steel is enough
Oil press: high steel for the internal mechanism, mild steel for the
casing

Commit:
Steel update: turret mount

Mild steel is a guess (it should be tough and not necessarily
hard)

Commit:
Steel update: tables

Workbench surface: mild steel is enough

Commit:
Steel update: solar

Mild steel is enough reinforcement for a fragile pane of glass

Commit:
Steel update: seating

These are lightly loaded parts, mild steel is enough

Commit:
Steel update: helicopter rotors

These are made out of medium steel:
https://www.reddit.com/r/AerospaceEngineering/comments/1bz8txj/which_steel_gradesalloys_are_used_in_helicopter/
https://www.ssa-corp.com/documents/Data%20Sheet%204340-AMS-6414-UNSG43406.pdf

Commit:
Steel update: vehicle rigs

Lightly loaded parts or just workbenches etc.: mild steel

Commit:
Steel update: ram

Has to be both hard (to shred zombies) and tough (to resist impacts): I
guesstimate medium steel. (Alternatively, it could be tempered steel, but it's
a big part, would it be even obtainable?)

Commit:
Steel update: plating

If the plating can be mild steel, I guess the spikes would be the same
grade.

Commit:
Steel update: noise

Lightly loaded parts: mild steel

Commit:
Steel update: motors

The steel used in electric motors is chosen for its magnetic properties
and is a mild steel , or rather a specialty steel with low carbon
content (https://en.wikipedia.org/wiki/Electrical_steel).
Mild steel is a simplification, but should be a decent one.

I also added copper to the motor materials, for obvious reasons.

Commit:
Steel upate: windmills

Steel wind turbines in US are made out of low-carbon steel:
https://www.leecosteel.com/news/post/exploring-steel-plate-used-in-the-american-wind-industry/

Commit:
Steel update: human-powered

Wheelchair hand rims: I guess mild steel (not highly loaded)
Bicycle cranks: I guess medium steel (higher loading, people do jump on
them)

Commit:
Steel update: lights

lightly loaded parts: mild steel

Commit:
Steel update: fuel storage

just a metal bin: mild steel

Commit:
Steel update: frames

Light frame: mild steel, just as all non-Xlight metal frames

Commit:
Steel update: farming

Scoop is described as made from sheet metal which is mild steel
Other tools (drill seeds, reapers, plows) are modelled after plows
which are generally from medium carbon "plow steel":
https://amsresources.com/1044-m1044-plow-steel-medium-carbon-steel/

Commit:
Steel update: engineering equipment

These are informed guesses.
Lifting equipment is generally made out of unhardened (i. e. mild) steel
to avoid dangerous sudden breaks.
Rockwheel is probably "plow steel": https://amsresources.com/1044-m1044-plow-steel-medium-carbon-steel/

Commit:
Steel update: engines

Switched the gas turbines from cast iron to superalloy. The real
turbines are made out of high-end stainless steels requiring complex heat
treatment that can't realistically be done in home shop.

Given that the small turbine is from 1960s, it could be closer to modern
tempered steel?

https://turbinebladesteel.com/turbine-blade-steels/
https://cdn.intechopen.com/pdfs/22905/InTech-Materials_for_gas_turbines_an_overview.pdf

Commit:
Steel update: controls

Generally lightly loaded parts: mild steel

Commit:
Steel update: cargo space

Vehicle parts: mild steel (similar to frames)
Locks: case-hardened steel (the real things are differentially hardened
IIUC, case-hardening is the closest analog

Commit:
Steel update: batteries

Battery cases are not heavily loaded: mild steel

Commit:
Steel update: makeshit vehicle armor

Based on the respective recipes

Commit:
Steel update: animal yokes

Commit:
Steel update: alternators

Alternators use a specialty steel https://en.wikipedia.org/wiki/Electrical_steel
but a mild steel should be a decent proxy

GuardianDll commented:
> Is there a way to test a lot of items at once?

Generally you do this changes locally, in which case you can either compile the game or, if you lack compiling tools, just grab the executable file from https://github.com/CleverRaven/Cataclysm-DDA/releases that roughly matches the version you are doing the changes, place it in the game folder, and boot it up

GuardianDll commented on data/json/items/vehicle/manual.json:
```diff
@@ -20,5 +20,6 @@
     "copy-from": "hand_rims",
     "name": { "str": "foot crank" },
     "description": "A pedal and gear assembly from a bicycle."
+    "material": [ "mc_steel" ],
```
```suggestion
    "description": "A pedal and gear assembly from a bicycle.",
    "material": [ "mc_steel" ]
```

GuardianDll commented on data/json/items/vehicle/engine.json:
```diff
@@ -344,6 +344,8 @@
     "description": "A gas turbine engine.  Used primarily for small helicopters.",
     "//": "Based on Allison 250-C30R.  Doubled the published weight to account for a transmission.",
+    "//2": "It's from 60s, tempered steel might be appropriate?"
+    "material": [ "superalloy" ],
```
Please refrain from using superalloy in mundane items, ideally it should be removed, and left only for mods like aftershock

Lycowolf replied:
So superalloy is sci-fi material from far future? Because some of the materials used for modern gas turbines / jet engines are literally named superalloys: see https://cdn.intechopen.com/pdfs/22905/InTech-Materials_for_gas_turbines_an_overview.pdf

I couldn't find any info on specific steel grades / alloys used in these engines, but using a tempered steel (that can just be made by in a small workshop) didn't feel right. Is there a high-end steel in the game, or is tempered steel a decent proxy for the game?

Lycowolf replied:
The Allison 250-C30R is a little different: given its age, I suspect the original materials are close to modern high-speed steels (strong, temperature-resistant) so I believe it might be a good fit for the game

GuardianDll replied:
Yeah, even if the name is the same, in dda superalloy was used a lot as catchall for random super material type used in whatever author wanted, including superalloy vehicle rams and superalloy dog harnesses. Which, of coursed, poisoned the term similarly to "survivor" line of stuff, so moving away from it is "generally good". You can, of course, add new "high-performance alloy" material and items to represent this thing better, but i don't think we really need to go this deep in our abstractions

Lycowolf replied:
I really don't want to add a new material type, that's way out of scope of what I am trying to do. (I just want to edit existing things so they would stop using `"steel"` material, and fixing obvious mistakes along the way.)

I converted the motors to tempered steel. The only superalloy vehicle stuff remaining is superalloy plating, and it works as is, so I am not touching it.

GuardianDll commented on data/json/items/vehicle/engine.json:
```diff
@@ -344,6 +344,8 @@
     "name": { "str": "25 L 650 HP gas turbine engine" },
     "description": "A gas turbine engine.  Used primarily for small helicopters.",
     "//": "Based on Allison 250-C30R.  Doubled the published weight to account for a transmission.",
+    "//2": "It's from 60s, tempered steel might be appropriate?"
```
It doesn't need to match the specs of exact item the stats are derived for, but if they are generally made out of tempered steel, yes, it makes total sense to use tempered steel material

Lycowolf commented:
Testing: I tried spawning every updated item, it worked OK except:
 - `wind_turbine` can't be spawned, it's not in the debug menu. `xl_wind_turbine` that's derived from it spawns fine. I reverted to the current master and hit the same issue, and my edits look OK to me, so I think it's a bug elsewhere

Commit:
Steel update: superalloy engines

Convert superalloy to tempered steel to avoid using superalloy for
mundane parts.
See https://github.com/CleverRaven/Cataclysm-DDA/pull/80646 for
rationale

XygenSS reviewed (commented):
Thanks for all the work, just minor nitpicks.
I'm not the authority on steel grades, so you should take other peoples' review (if there are any) over mine.

Lycowolf commented:
> Thanks for all the work, just minor nitpicks. I'm not the authority on steel grades, so you should take other peoples' review (if there are any) over mine.

I am definitely not an authority either. I tried to justify my decisions in commit messages, seems like it's not readily visible. I'll add explanatory comments to the jsons.

Lycowolf commented on data/json/items/vehicle/cargo.json:
```diff
@@ -40,7 +40,7 @@
     "symbol": "+",
     "color": "dark_gray",
-    "material": [ "steel" ],
+    "material": [ "ch_steel" ],
```
I imagined something like a padlock, and good padlocks have case hardened cases and shackles (e. g. https://americanlocks.com/products/a5360).

But reading your comment, I see these are probably car locks, will change to mild steel.

XygenSS replied:
even in padlocks, only the outer body is designed to withstand abuse, the inner core is made of a comparatively softer material that is easier to machine and mass produce. Drill protection is achived by insertion of tiny hardened steel pins into the core.

GuardianDll replied:
We can use two materials to represent it

Lycowolf commented on data/json/items/vehicle/engine.json:
```diff
@@ -344,6 +344,7 @@
     "name": { "str": "25 L 650 HP gas turbine engine" },
     "description": "A gas turbine engine.  Used primarily for small helicopters.",
     "//": "Based on Allison 250-C30R.  Doubled the published weight to account for a transmission.",
+    "material": [ "qt_steel" ],
```
See the tread https://github.com/CleverRaven/Cataclysm-DDA/pull/80646#discussion_r2051786921

Correct solution would be to probably split superalloy to a aerospace-grade steel and sci-fi alloy, but that's definitely out of scope.

At least the Allison 250-C30R seems to be older than modern superalloys, so I imagined the casing and turbine blade materials would be close to modern HSS (i. e. strong and heat-resistant steel) and used tempered steel as a proxy. But frankly, I couldn't find anything and this is just a guess.

Lycowolf replied:
>  But frankly, I couldn't find anything and this is just a guess.

I mean, I read the links below but it's too new and shows that the materials used are modern high-end steels that don't look like something you can work with in a small workshop that fits into your car.

https://turbinebladesteel.com/turbine-blade-steels/
https://cdn.intechopen.com/pdfs/22905/InTech-Materials_for_gas_turbines_an_overview.pdf

GuardianDll replied:
you can add a comment that it needs a material that represents it better

Lycowolf commented on data/json/items/vehicle/farming.json:
```diff
@@ -67,7 +67,7 @@
     "description": "A heavy assembly of wheels and steel blades that turn up the ground.",
     "weight": "32666 g",
-    "material": [ "steel", "plastic" ],
+    "material": [ "mc_steel", "plastic" ],
```
I believe at least some (if not most) of them are made out of "plow steel" which is a medium steel.
https://amsresources.com/1044-m1044-plow-steel-medium-carbon-steel/

Lycowolf replied:
also https://www.bladeforums.com/threads/anybody-know-what-type-of-steel-plow-steel-might-be.1073687/

Lycowolf commented on data/json/items/vehicle/farming.json:
```diff
@@ -20,7 +20,7 @@
     "name": { "str": "seed drill" },
     "description": "An assembly of tubes, spikes, and wheels that, when dragged along the ground, allows a vehicle to plant seeds automatically in suitably tilled land.",
-    "material": [ "steel", "plastic" ],
+    "material": [ "mc_steel", "plastic" ],
```
I imagined they would use the same steel as plows (which might be wrong)
I searched it but had no luck, can you please link source?

Lycowolf commented on data/json/items/vehicle/farming.json:
```diff
@@ -29,7 +29,7 @@
     "name": { "str": "reaper" },
     "description": "An assembly of a blade, wheels, and a small lever for engaging/disengaging, used to cut down crops prior to picking them up.",
-    "material": [ "steel", "plastic" ],
+    "material": [ "mc_steel", "plastic" ],
```
same as with seed drills

Lycowolf commented on data/json/items/vehicle/manual.json:
```diff
@@ -19,6 +19,7 @@
     "name": { "str": "foot crank" },
-    "description": "A pedal and gear assembly from a bicycle."
+    "description": "A pedal and gear assembly from a bicycle.",
+    "material": [ "mc_steel" ]
```
I found nothing on the specific steel used, but as I see it, the loading is similar to a wheel axles that are made out of medium steel

Lycowolf commented on data/json/items/vehicle/motors.json:
```diff
@@ -21,7 +21,7 @@
     "description": "A powerful electric motor, as found in those consumer electric sports cars with ridiculous or ludicrous acceleration options.",
     "weight": "120 kg",
-    "material": [ "superalloy", "copper" ],
+    "material": [ "qt_steel", "copper" ],
```
OK

Lycowolf commented on data/json/items/vehicle/plating.json:
```diff
@@ -174,7 +174,7 @@
     "color": "dark_gray",
     "symbol": "]",
-    "material": [ "steel" ],
+    "material": [ "lc_steel" ],
```
yes, cast iron sounds like a god fit

Lycowolf commented on data/json/items/vehicle/rams.json:
```diff
@@ -10,7 +10,7 @@
     "price_postapoc": "25 USD",
     "to_hit": -1,
-    "material": [ "steel" ],
+    "material": [ "mc_steel" ],
```
I am pretty sure mild steel wouldn't cut it: it can't hold an edge and would deform when grinding bones.

Recycling shredders (which I imagined this thing to be like) use tool steels or carbide blades (https://maxtormetal.com/shredder-blades-material-life/), but that seemed too hard to get, so I substituted a medium steel, as it cans still be found in larger quantities (in vehicle axles etc.)

Lycowolf replied:
> The bigger issue is that the shredder in the game doesn't really make sense. For one, it is described as being rotated by the engine, but the part doesn't have any such interaction in the JSON (and could hypothetically be mounted on a canoe).

I suggest we keep the scope manageable. Originally, I wanted to update everything under `items/` but it seems overwhelming task now even as is.

Also, now I want a zombie shredder on a unicycle ;-)

Lycowolf commented on data/json/items/vehicle/wheel.json:
```diff
@@ -25,7 +25,7 @@
     "price": "10 cent",
     "price_postapoc": "10 cent",
-    "material": [ "steel" ],
+    "material": [ "mc_steel" ],
```
My (limited) sources say that 1040 (which is a medium steel) and similar steels are common:
https://www.iforgeiron.com/topic/4526-axles-what-metal/

Lycowolf commented on data/json/items/vehicle/engineering.json:
```diff
@@ -49,7 +49,7 @@
     "description": "A large, heavy, jagged metal disc for digging trenches.",
     "weight": "62000 g",
-    "material": [ "steel" ],
+    "material": [ "mc_steel" ],
```
So they can be made out of (basically) automotive steel, which is low-carbon steel in game? That sounds appropriate for something survivors would actually use.

Commit:
Steel update: review fixes

Updated some of them materials and added comments where reviewers had questions.

Maleclypse committed:
Merge branch 'master' into steel-update

Commit:
Steel update: add comment about turbine materials

Lycowolf commented:
I consider this in a good shape. If you see issue with it, please tell me, otherwise please merge it at your convenience.

GuardianDll commented:
I agree, there may be issues here and there, but they can be adressed in future PRs
